sympy/functions/elementary/exponential.py

```diff
@@ -456,6 +456,19 @@ def _eval_rewrite_as_tanh(self, arg):
         from sympy import tanh
         return (1 + tanh(arg/2))/(1 - tanh(arg/2))
 
+    def _eval_rewrite_as_sqrt(self, arg):
+        from sympy.functions.elementary.trigonometric import sin, cos
+        if arg.is_Mul:
+            coeff = arg.coeff(S.Pi*S.ImaginaryUnit)
+            if (coeff == 0):
+                return None
+            else:
+                cosine, sine = cos(S.Pi*coeff), sin(S.Pi*coeff)
```

Comment:
These can be numbers only if `coeff` is a (non-zero) number. That could be checked by `if coeff and coeff.is_number`. That would probably make this more efficient by avoiding useless calls of `cos` and `sin` .